src/components/profile/profileComponents/ProfileDetails.js

```diff
@@ -0,0 +1,190 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import role from '../../../assets/images/role.png';
+import dob from '../../../assets/images/dob.png';
+import email from '../../../assets/images/email.png';
+import gender from '../../../assets/images/gender.png';
+import profile from '../../../assets/images/profile.png';
+import language from '../../../assets/images/language.png';
+import currency from '../../../assets/images/currency.png';
+import departement from '../../../assets/images/departement.png';
+import backgroundProfile from '../../../assets/images/backgroundProfile.png';
+import { fetchUserProfileDetails, updateUserProfileDetails } from '../../../actions';
+import { shortData, componentDidMountHelper, checkForm } from '../../../helpers/ProfileHelper/profileHelper';
+
+
+class ProfileDetails extends Component {
+  constructor() {
+    super();
+    this.state = {
+      isUpdateFormShown: false,
+      name: '',
+      gender: '',
+      birthdate: '',
+      preferredLanguage: '',
+      preferredCurrency: '',
+      location: '',
+      inputError: '  ',
+    };
+  }
+
+  handleChange(key) {
+    this.setState({ [key.target.id]: key.target.value });
+  }
+
+  componentDidMount() {
+    const {
+      token,
+      username
+    } = componentDidMountHelper();
+    this.props.fetchThisUserDetails(
+      token,
+      username
+    );
+  }
+
+  onsubmit() {
+    const { token, username } = componentDidMountHelper();
+    const formError = checkForm(this.state);
+    this.setState({ inputError: formError });
+    if (formError === 'Profile updated successfully') {
+      this.props.updateThisUserDetails(token, username, this.state);
+      window.location.reload(true);
```

Comment:
Can you use redirect instead of window.location.reload. If you are updating the user profile simply to a re-direction

Reply:
Noted TTL @luc-tuyishime.